add-on/chrome/test/mochitest/browser_copypanel.js

```diff
@@ -0,0 +1,114 @@
+/* Any copyright is dedicated to the Public Domain.
+   http://creativecommons.org/publicdomain/zero/1.0/ */
+
+"use strict";
+
+let gConstants;
+LoopAPI.inspect()[1].GetAllConstants({}, constants => gConstants = constants);
+
+let { goDoCommand, gURLBar } = window;
+
+registerCleanupFunction(() => {
+  document.getElementById("loop-notification-panel").hidePopup();
+  LoopUI.removeCopyPanel();
+});
+
+// Even with max threshold, no panel if already shown.
+add_task(function* test_init_copy_panel_already_shown() {
+  LoopUI.removeCopyPanel();
+  Services.prefs.setIntPref("loop.copy.ticket", -1);
+  Services.prefs.setBoolPref("loop.copy.shown", true);
+
+  yield LoopUI.maybeAddCopyPanel();
+
+  Assert.equal(document.getElementById("loop-copy-notification-panel"), null, "copy panel doesn't exist already shown");
+  Assert.equal(Services.prefs.getIntPref("loop.copy.ticket"), -1, "ticket should be unchanged");
+});
+
+// Even with max threshold, no panel if private browsing.
+add_task(function* test_init_copy_panel_private() {
+  LoopUI.removeCopyPanel();
+  Services.prefs.setIntPref("loop.copy.ticket", -1);
+
+  let win = OpenBrowserWindow({ private: true });
+  yield new Promise(resolve => win.addEventListener("load", resolve));
+  yield win.LoopUI.maybeAddCopyPanel();
+
+  Assert.equal(win.document.getElementById("loop-copy-notification-panel"), null, "copy panel doesn't exist for private browsing");
+  Assert.equal(Services.prefs.getIntPref("loop.copy.ticket"), -1, "ticket should be unchanged");
+
+  win.close();
+});
+
+// Helper function for testing clicks on the copy panel.
+function testClick(domain, onIframe) {
+  let histogram = Services.telemetry.getHistogramById("LOOP_COPY_PANEL_ACTIONS");
+  histogram.clear();
+  LoopUI.removeCopyPanel();
+  gURLBar.value = "http://" + domain + "/";
+
+  LoopUI.addCopyPanel();
+  gURLBar.focus();
+  gURLBar.select();
+  goDoCommand("cmd_copy");
+
+  let iframe = document.getElementById("loop-copy-panel-iframe");
+  return new Promise(resolve => {
+    iframe.addEventListener("DOMContentLoaded", () => {
+      iframe.parentNode.addEventListener("popuphidden", () => resolve(histogram));
+      setTimeout(() => onIframe(iframe));
```

Comment:
Let's add a comment explaining why calling without an interval is necessary here.